### DISCUSSION TITLE
有没有generate的库

### DISCUSSION THREAD

**Discussion by dhlolo** [wxmlfile/wxml-parser#16] [Ideas]
在项目中分析ast之后，在小程序中我们会做一些压缩的操作，因此需要transform和generate，但似乎现在这方面还比较欠缺。因为这个原因，我们项目中目前还是用htmlparser2和单独提取wxml{{}}中表达式的方式来做，不过ast的解析确实没有这个parser做得好，所以想问作者有没有这方面的计划？

**Comment by iChenLei** [comment]
这个库的定位是用做`eslint-plugin`的`custom wxml parser`, 所以一开始就没有考虑做成一个类似`babel-for-wxml`的库，codegen和transform我没有很强的需求和动力去做。

**Reply by dhlolo** [reply]
你们没有小程序主包压缩的需求么？这方面是怎么实践的，有可以分享的么？

**Reply by iChenLei** [reply]
对于`js`和`wxss`的压缩应该无需我多说，方案和库社区很多，我想你主要是想交流`wxml`的压缩。这方面我们做的比较简单，暴力用正则来做的，需要注意的是需要隔离出`<wxs>`的内容，然后拼接字符串。
> 再说这个库开发的动机，小程序项目除了wxml别的文件诸如js/wxss(可视为css)都有相关的lint工具可以保证代码质量，所以就基于eslint做了个wxml的plugin，这个库是eslint-plugin-wxml的衍生需求

**Reply by dhlolo** [reply]
感谢。除了app.js以及其引用的js可以通过很大参考一般的web方案之外。大部分js和wxss的内容在压缩时，如果需要进一步在miniprogram-ci的压缩配置之上再做一些，比如wxss的类名压缩；js中(ComponentJS，PageJS，BehaviorJS)的类方法，数据和属性压缩，往往也需要解析其关联的wxml中的字符并替换来做压缩。
